webpack/assets/javascripts/react_app/components/ReportTemplatesTable/ReportTemplatesTable.js

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect } from 'react';
+import { Spinner } from 'patternfly-react';
+import PropTypes from 'prop-types';
+import { Table } from '../common/table';
+import { STATUS } from '../../constants';
+import MessageBox from '../common/MessageBox';
+import { translate as __ } from '../../common/I18n';
+import createReportTemplatesTableSchema from './ReportTemplatesTableSchema';
+import { getURIQuery } from '../../common/helpers';
+
+const ReportTemplatesTable = ({
+  getTableItems,
+  sortBy,
+  sortOrder,
+  error,
+  status,
+  results,
+}) => {
+  useEffect(() => {
+    getTableItems(getURIQuery(window.location.href));
+  }, [getTableItems]);
+
+  if (results.length === 0) {
+    return <Spinner size="lg" loading />;
+  }
+
+  if (status === STATUS.ERROR) {
+    return (
+      <MessageBox
+        key="report-templates-table-error"
+        icontype="error-circle-o"
+        msg={__(`Could not receive data: ${error && error.message}`)}
+      />
+    );
+  }
+
+  const availableActions = {
+    generate: (templateId, templateName) => {
```

Comment:
nit: unused argument `templateName`

since the actions are passed through schema to cell formatter, maybe the declaration can be moved to `ReportTemplatesTableSchema`, but leaving them here is also ok.